### ISSUE TITLE
Global form errors

### ISSUE THREAD

**Issue by arkwright** [erikras/redux-form#72]
redux-form does not seem to support global form errors (as opposed to field errors). For example, if the server responds with a 500 error, it is essential that the user be informed of this problem. But this error doesn't really apply to a _field_, per se.

I am currently hacking this functionality together via the `formReducer.plugin()` API, which is injecting a custom `errors` property into every form's state. I am then dispatching a custom `FORM_ERRORS` event with `form` (name) and `errors` (array of error strings) attached. My plugin listens for this action and merges in the errors.

Is there a better way to accomplish this using redux-form built-ins? If not, I propose that global form errors be adopted as a feature. My first thought is that a `redux-form/FORM_ERRORS` action could be exposed, which could be dispatched when the server response is received.

**Comment by erikras** [comment]
I have been doing it separately, as I don't think it's any of redux-form's business what or how the form gets submitted, which means that you've got to have a separate action/reducer ([duck](https://github.com/erikras/ducks-modular-redux)) to handle that, which means the errors can just be kept there. That is an interesting use of the `plugin()` api, but I'm not sure how much value is gained by having it inside the form reducer instead of in the reducer that the submit action belongs to. Does that make sense? What do you think?

**Comment by arkwright** [comment]
My feeling is that there ought to be an idiomatic way to render a server error using any given form framework. If this isn't automated within the framework to reduce boilerplate, then some sort of recipe should be made available so that users don't get caught reinventing the wheel.

I feel that storing the global form error state within the `form[formname]` property is the correct location. Otherwise we end up with global form errors stored as a sibling, e.g.:

```
{
  form: formReducer,
  globalFormErrors: globalFormErrorsReducer
}
```

And then the user has to manually `connect()` (I am using `.plugin()` to avoid manually connecting), or pass down the global form errors as props. This becomes quite a bit of boilerplate and "advanced" technique for a use case that every form will need to address.

You mentioned that it isn't "any of redux-form's business what or how the form gets submitted". Granting this, redux-form _does_ already accept a promise returned by `onSubmit` which expects server-side field errors in a specific format. I don't think it is a semantic stretch to add a `_global` property (or something) to the error object returned, and make these errors available via `this.props` for the user so that the form can render them. However, this does create two-way data flow with respect to errors, and so ultimately I feel that errors (both form and field) ought to end up in the store.

I'm not sure if that helps to clarify. Let me know how you feel.

**Comment by erikras** [comment]
Yes, you're making a good argument. I'm convinced.

**Comment by arkwright** [comment]
:+1: Thanks for your consideration!

**Comment by erikras** [comment]
Fixed in [v1.6.0](https://github.com/erikras/redux-form/releases/tag/v1.6.0).

**Comment by arkwright** [comment]
Wow, that was fast. Nice work! Thank you! I will try this out ASAP.
